subsys/net/Kconfig

```diff
@@ -15,7 +15,7 @@ if NET_BUF
 
 config NET_BUF_USER_DATA_SIZE
 	int "Size of user_data available in every network buffer"
-	default 8 if (BT || NET_TCP2) && 64BIT
+	default 10 if (BT || NET_TCP2) && 64BIT
```

Comment:
Setting the default does not ensure correctness. I suggest adding a `depends on NET_BUF_USER_DATA_SIZE >= 0` in the BT_ISO Kconfig as well.

Reply:
This must be broken on non 64BIT systems then. And instead if increasing the USER_DATA_SIZE I suggest to do what we do for HCI commands instead.

```
static struct cmd_data cmd_data[CONFIG_BT_HCI_CMD_COUNT];

#define cmd(buf) (&cmd_data[net_buf_id(buf)])
```

Reply:
I simply added 2 octets to the already 8 octets of user data for ISO packets - and tracked the user-data size back to this define. It might not be the right one, but if it is not, I would appreciate a pointer to how the current 8 bytes of user-data is configured.
I don't understand this reference to hci commands - for HCI commands you save some context information about HCI commands that are being processed, where for ISO data you put the headers of the ISO data you receive from the controller into the user-data of the net_buf you pass on the application layer.

Reply:
@Casper-Bonde-Bose Sorry for not explaining properly. My point was since user data of ISO was 8 bytes, and the default user data length for when `64BIT=n` then there must be an existing memory bug on 32-bit systems.
And since increasing the user-data size of the net bufs would affect the sizes of all the buffers in the system then that is quite an expensive.

To avoid this we have used a different approach for the HCI commands, by declaring an array of user_data, and using the
net_buf_id to index it.
Which should end up something like this then:
```
static struct iso_data iso_data[CONFIG_BT_ISO_RX_BUF_COUNT];
#define iso(buf) (&iso_data[net_buf_id(buf)])
```

Reply:
Agreed; since this is used to hold the `struct iso_data` (i.e. ISO specific) I think a separate but "linked" array in iso.c is the better solution overall (will also be more future proof than incrementing the user_data size).

Reply:
I ran into this issue when developing the iso adaptation layer (currently in draft), I added the following 
default 8 if ((BT || NET_TCP2) && 64BIT) || BT_ISO

This works for me while still using the userdata in the net_buf

Reply:
@asmk-ot Yes, that would work for the current 8 bytes ISO user_data, but I extended to 10 bytes in this change. I think the issue people have with increasing this value is that all subsystems that use the net_buf system will bet a mem-penelty of 8 bytes per entry - hence the solution will be changed to use only the default 4 bytes of user_data and move the iso-specific data into a separate array.
By the way you will see all sorts of odd behavior on 32 bit systems when you actually start to receive ISO data. When you reach the last buffer in the pool you will overwrite what ever 4 bytes of memory the compiler/linker placed after the iso-pool array.